doc/modules/mod_global_distrib.md

```diff
@@ -0,0 +1,233 @@
+Client's process binds itself to a connection to a remote datacenter on first use, and henceforth always uses this connection to route messages directed to this datacenter.
+This - along with the dedicated worker process on the receiver's side - ensures that simple cross-datacenter messages between two entities are delivered in their sending order.
+
+It may happen that a message is rerouted through multiple datacenters (e.g. if the user has reconnected to a different datacenter while the message was already in flight).
+Messages are given a TTL parameter by the source datacenter so that they cannot be rerouted indefinitely.
+The TTL is decreased on each reroute.
+Note that in the edge case of multi-datacenter routing, the messages may be received out-of-order at the destination datacenter.
+
+#### Bounce
+
+Consider the following edge case: user **U1** logged into datacenter **DC2** quickly reconnects to datacenter **DC3**.
+Because session table has not yet been replicated, **DC2** does not see **U1** in the session table, while a different datacenter **DC1** still sees **U1** logged into **DC2**.
+When **U2**, logged into **DC1**, sends a message to **U1**, it will now be rerouted to **DC2** even though the user is now available at **DC3**.
+
+![State after U1's reconnection](mod_global_distrib_bounce_example.svg)
+
+Bounce mechanism solves this and similar edge cases by storing messages for which there is no known routing in the current datacenter.
+The stored messages are then assigned a bounce-TTL value and periodically - with backoff - are attempted to be routed again.
+In the example above, the message from **U2** would be temporarily stored at **DC2** and rerouted successfully once **DC2** learns (via replication) that **U1** is available at **DC3**.
+
+> Note: bounce mechanism, similarly to multi-datacenter routing, may result in out-of-order messages being received at the destination datacenter.
+
+#### Metrics
+
+Global distribution modules expose several per-datacenter metrics that can be used to monitor health of the system. All metrics begin with **global.mod_global_distrib** prefix:
+
+* `outgoing.messages.<host>`: number of cross-datacenter messages sent by this cluster to a given host.
+* `incoming.messages.<host>`: number of cross-datacenter messages received by this cluster from a given host.
+* `incoming.transfer_time.<host>` *[us]*: time elapsed between sending and receiving the message over the network from a given host.
+  The duration is calculated using wall clock times on sender and receiver node.
+* `outgoing.queue_time.<host>` *[us]*: time elapsed while message waits in queue of sender connection to a given host.
```

Comment:
time elapsed while message waits in a queue of a sender connection to a given host.